src/batch_processing/calc_dupe_candidates.py

```diff
@@ -0,0 +1,46 @@
+import sys
+import os
+from pyspark.sql.types import FloatType
+from pyspark.sql.functions import col, udf
+
+from pyspark.conf import SparkConf
+from pyspark.context import SparkContext
+from pyspark.sql import SQLContext, DataFrame
+
+global sc
+spark_conf = SparkConf()
+sc = SparkContext(conf=spark_conf)
+sc.setLogLevel("ERROR")
+sc.addFile(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/lib/min_hash.py")
+sc.addFile(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/lib/locality_sensitive_hash.py")
+sc.addFile(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/lib/util.py")
+sc.addFile(os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/config/config.py")
+
+global sql_context
+sql_context = SQLContext(sc)
+
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/config")
+sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))) + "/lib")
+
+import config
+import util
+import min_hash
+import locality_sensitive_hash
+import configparser
+
+def common_bands_count(a, b):
```

Comment:
Each of these processing steps/python files should be should be scoped so that someone does not accidently run them by importing them. Each of them should have the commands in `if __name__ == “__main__”:` loop to prevent that. I also suggest that the main loop only load the configs and arguments to the script. the spark context and ETL functions can then be in a class that that you call in the the main loop. Doing this will let your code be extensible if you need to call this processing step it in another python script.

Reply:
@tyaq agreed. With compare.py I added this in and will enforce it for the other files.